realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -355,12 +360,7 @@ JNIEXPORT void JNICALL Java_io_realm_internal_TableQuery_nativeBetween__J_3JFF(
             Q(nativeQueryPtr)->between(S(arr[0]), static_cast<float>(value1), static_cast<float>(value2));
         }
         else {
-            Q(nativeQueryPtr)->group();
-            Table* tbl = getTableLink(nativeQueryPtr, arr, arr_len);
-            Q(nativeQueryPtr)->and_query(numeric_link_greaterequal<Float, float, jfloat>(tbl, arr[arr_len-1], value1));
-            tbl = getTableLink(nativeQueryPtr, arr, arr_len);
-            Q(nativeQueryPtr)->and_query(numeric_link_lessequal<Float, float, jfloat>(tbl, arr[arr_len-1], value2));
-            Q(nativeQueryPtr)->end_group();
+            ThrowException(env, IllegalArgument, "between does not support link queries.");
```

Comment:
YEAH... I would be difficult to pick this one out of the PR... I can create a test case for this after this pr got merged.